build.rs

```diff
@@ -127,6 +146,19 @@ fn configured_by_vcpkg() -> bool {
     false
 }
 
+fn use_general_lib_dir() -> Result<String, VarError>{
```

Comment:
I would change this function in such a way that it accepts the name of a environment variable as input. This would enable reusing the same function for target specific variables as well and unify even more code there.